WordPress/Classes/Models/AbstractPost.m

```diff
@@ -583,6 +592,32 @@ - (void)markRemoteStatusFailed
     [self save];
 }
 
+- (BOOL)shouldAttemptAutoUpload {
+    if (!self.confirmedChangesTimestamp || !self.confirmedChangesHash) {
+        return NO;
+    }
+
+    NSTimeInterval timeDifference = [[NSDate date] timeIntervalSinceDate:self.confirmedChangesTimestamp];
+
+    BOOL timeDifferenceWithinRange = timeDifference <= (60 * 60 * 24 * 2);
+    // I know, I know. This isn't how a date comparison _should_ be done!
```

Comment:
I know it's pretty obvious but can you add a comment saying something like "Post that is older than 48 hours will not be auto upload) ?
I would personally also get rid of the current comment. And instead just say that this method can potentially be called multiple times.